src/Workspaces/CSharp/Portable/Simplification/CSharpSimplificationService.Expander.cs

```diff
@@ -978,6 +979,38 @@ public override SyntaxNode VisitInvocationExpression(InvocationExpressionSyntax
                 // This Annotation copy is for the InvocationExpression
                 return originalNode.CopyAnnotationsTo(replacementNode).WithAdditionalAnnotations(Simplifier.Annotation);
             }
+
+            public override SyntaxNode VisitAnonymousObjectMemberDeclarator(AnonymousObjectMemberDeclaratorSyntax node)
+            {
+                // Expands "var a = new { A.B.C }" into "var a = new { C = A.B.C }"
+
+                var rewrittenNode = base.VisitAnonymousObjectMemberDeclarator(node) as AnonymousObjectMemberDeclaratorSyntax;
+                if (rewrittenNode.NameEquals != null)
+                {
+                    return rewrittenNode;
+                }
+
+                // Try to get the implicit property name from the expression
+                string implicitPropertyName = null;
+
+                var identifier = rewrittenNode.Expression as IdentifierNameSyntax;
+                var memberAccess = rewrittenNode.Expression as MemberAccessExpressionSyntax;
+                if (identifier != null)
+                {
+                    implicitPropertyName = identifier.Identifier.Text;
+                }
+                else if (memberAccess != null)
+                {
+                    implicitPropertyName = memberAccess.Name?.Identifier.Text;
+                }
+
+                if (implicitPropertyName == null)
+                {
+                    return rewrittenNode;
+                }
+
+                return rewrittenNode.WithNameEquals(SyntaxFactory.NameEquals(implicitPropertyName).WithAdditionalAnnotations(Simplifier.Annotation, Formatter.Annotation));
```

Comment:
total nit: Maybe break this onto two lines (placing SyntaxFactory.NameEquals on its own line) to make it clearer what the target of ```WithAdditionalAnnotations``` is?